synfig-studio/src/gui/canvasview.cpp

```diff
@@ -1543,6 +1534,35 @@ CanvasView::init_menus()
 
 }
 
+void
+CanvasView::init_plugins()
+{
+	auto instance = get_instance().get();
+	if(instance)
+		for ( const auto& plugin : App::plugin_manager.plugins() )
+		{
+			std::string id = plugin.id;
+			if(!action_group->get_action(id))
+				action_group->add(
+					Gtk::Action::create(id, plugin.name.get()),
+					[instance, id](){instance->run_plugin(id, true);}
```

Comment:
The signal slot here should use[ `sigc::track_obj`](https://libsigcplusplus.github.io/libsigcplusplus/reference/html/group__track__obj.html#gad4baa7c8ace9d51344f85d121d08b2cc).
It's important to autodelete signal connection if instance object is deleted first, for example.

![image](https://github.com/user-attachments/assets/625877cf-a2d6-449a-9b86-afd14d85e3cd)

(the current docs are for sigc++ 3.0 API. Sadly libsigc++ website doesn't host docs for API 2.0 anymore)

Reply:
To make myself clear: My comment above was related to lambda function as a signal slot.

I built and uploaded the docs for libsigc++ 2.12 and gtkmm 3.24 :
http://floss.rodolfo.eng.br/libsigc++/2.12.1/
http://floss.rodolfo.eng.br/gtkmm/3.24.4/